test/constraint.jl

```diff
@@ -377,6 +377,31 @@ end
 
 @testset "Constraints for JuMP.Model" begin
     constraints_test(Model, JuMP.VariableRef)
+    @testset "all_constraints (scalar)" begin
+        model = Model()
+        @variable(model, x >= 0)
+        ref = all_constraints(model, VariableRef, MOI.GreaterThan{Float64})
+        @test ref == [LowerBoundRef(x)]
+        aff_constraints = all_constraints(model, AffExpr,
+                                          MOI.GreaterThan{Float64})
+        @test isempty(aff_constraints)
+        # GreaterThan isn't a concrete type.
+        @test_throws ErrorException all_constraints(model, AffExpr,
```

Comment:
Since we dropped Julia v0.6, we can now do
```julia
err = ErrorException("MOI.GreaterThan is not a concrete type. Did you miss a type parameter?")
@test_throws err ...
```
To make sure that this error is thrown instead of another one